src/traces/histogram/calc.js

```diff
@@ -512,3 +510,19 @@ function cdf(size, direction, currentBin) {
         }
     }
 }
+
+function calcSelection(cd, trace) {
+    if(Array.isArray(trace.selectedpoints)) {
+        var ptNumber2cdIndex = {};
+
+        // make histogram-specific pt-number-to-cd-index map object
```

Comment:
don't we want this available for use later, ie without a recalc? I would have imagined creating this in the [main binning loop](https://github.com/plotly/plotly.js/blob/315e6327bbd512b20cec61e651f6bb945c1521d0/src/traces/histogram/calc.js#L128) and stashing it in the full trace, also possibly creating it as an array rather than a map, seems like that would be more efficient and would work just as well?

Reply:
> don't we want this available for use later, ie without a recalc?

I don't think so. Calling `restyle` with `selectedpoints` has to trigger a recalc. On user selections, the calcdata items are mutated [directly](https://github.com/plotly/plotly.js/blob/261387dcb04502b82634e2ea220d3d828d3d3f8e/src/traces/bar/select.js#L33) w/o needing to refer back to input data-array indices. Am I missing something?

Reply:
> I would have imagined creating this in the main binning loop 

Good call, done in -> 7543dc6

Reply:
> Calling `restyle` with `selectedpoints` has to trigger a recalc.

hmm ok, I was hoping we could avoid that but we can discuss optimizing that later.